src/react/hooks/paragon/useParagonThemeCore.js

```diff
@@ -0,0 +1,150 @@
+import { useEffect, useState } from 'react';
+
+import { logError, logInfo } from '../../../logging';
+import { fallbackThemeUrl, removeExistingLinks } from './utils';
+
+/**
+ * Custom React hook that manages the loading and updating of the core Paragon theme CSS and the brand override
+ * theme CSS. It ensures that the core theme CSS (both default and brand override) is added to the document
+ * `<head>` as `<link>` elements.
+ *
+ * The function logs and handles fallback logic in case the core theme fails to load.
+ *
+ * @memberof module:React
+ *
+ * @param {Object} args - The arguments object containing theme and callback information.
+ * @param {Object} args.themeCore - The core theme configuration.
+ * @param {string} args.themeCore.urls.default - The URL to the default core theme CSS.
+ * @param {string} [args.themeCore.urls.brandOverride] - The URL to the brand override theme CSS (optional).
+ * @param {Function} args.onComplete - A callback function that is called once both the core Paragon (default)
+ * theme and brand override theme (if provided) are complete.
+ */
+const useParagonThemeCore = ({
+  themeCore,
+  onComplete,
+}) => {
+  const [isParagonThemeCoreComplete, setIsParagonThemeCoreComplete] = useState(false);
+  const [isBrandThemeCoreComplete, setIsBrandThemeCoreComplete] = useState(false);
+
+  useEffect(() => {
+    // Call `onComplete` once both the paragon and brand theme core are complete.
+    if (isParagonThemeCoreComplete && isBrandThemeCoreComplete) {
+      onComplete();
+    }
+  }, [isParagonThemeCoreComplete, isBrandThemeCoreComplete, onComplete]);
+
+  useEffect(() => {
+    // If there is no config for the core theme url, do nothing.
+    if (!themeCore?.urls) {
+      setIsParagonThemeCoreComplete(true);
+      setIsBrandThemeCoreComplete(true);
+      return;
+    }
+    const getParagonThemeCoreLink = () => document.head.querySelector('link[data-paragon-theme-core="true"]');
+    const existingCoreThemeLink = document.head.querySelector(`link[href='${themeCore.urls.default}']`);
+    const brandCoreLink = document.head.querySelector(`link[href='${themeCore.urls.brandOverride}']`);
+    if (!existingCoreThemeLink) {
```

Comment:
This is a bit of a style nit, but having so much nested in this `if` statement is making this a bit hard to follow for me. Would inverting the condition and using an early return work here?

Something like:
```js
if (existingCoreThemeLink) {
  existingCoreThemeLink.rel = 'stylesheet';
  existingCoreThemeLink.removeAttribute('as');
  existingCoreThemeLink.dataset.paragonThemeCore = true;
  if (brandCoreLink) {
    brandCoreLink.rel = 'stylesheet';
    brandCoreLink.removeAttribute('as');
    brandCoreLink.dataset.brandThemeCore = true;
  }
  setIsParagonThemeCoreComplete(true);
  setIsBrandThemeCoreComplete(true);
  return;
}

const getExistingCoreThemeLinks = [ ... ]
```

Reply:
Updated here https://github.com/openedx/frontend-platform/pull/689/commits/b5069bafc06a3ffc4321310acfd8727478c08952